platforms/platforms.go

```diff
@@ -223,6 +223,9 @@ func Parse(specifier string) (specs.Platform, error) {
 		}
 		if isKnownArch(p.Architecture) {
 			p.OS = runtime.GOOS
+			if p.OS == "windows" {
+				p.OSVersion = GetWindowsOsVersion()
```

Comment:
I think we should rethink some of this; Parse should really do _just that_ parse what's in the string format, and return the results.

We could consider a functional argument to provide defaults, but it's problematic that we start to fill in missing information; doing so means that we make assumptions based on the local platform , which can be problematic. See the discussion below for more context;
https://github.com/moby/buildkit/pull/4315#discussion_r1378282028

We should also formalize the string format; "/variant" (optional) and "/os-version" (also optional) can be ambiguous, which means we won't be able to tell for sure what's what.

/cc @dmcgowan @tonistiigi

Reply:
Also related; I once started a draft where I looked at how we currently handle some of this (but never found time to finish); here's a gist of some of the notes I drafted; https://gist.github.com/thaJeztah/a970f7b93f5f0a68b36b5dbb755c320d

Reply:
This looks like something done in the rest of the cases in the function now. Is this something that needs to be solved for this PR or more wholistically?